src/api.ts

```diff
@@ -210,6 +212,41 @@ export interface MongoDBOIDCPlugin {
    * for automatic token refreshing.
    */
   destroy(): Promise<void>;
+
+  /**
+   * Retrieve userinfo for the current token set subject
```

Comment:
Let’s also document that these methods _do not_ attempt to procure a valid token set before returning, that’s a bit surprising given that the plugin is supposed to do state management like that _for_ the user.

Reply:
Why is it surprising? It should be possible to try to get userinfo from any state, whether or not your token set is valid, if it's not valid or missing, backend should handle this and return error

Reply:
Well, like I said, the plugin is supposed to do the state management here. If you call a method on it that makes use of tokens, I’d always expect it to first make sure that the current token is up-to-date and do a background refresh if it’s not. (That doesn’t mean that that has to _succeed_, it could fail to refresh and we could still call the userinfo endpoint in that case.)

Not saying that this needs to change in the docs, but yes, should definitely be documented.

Reply:
My main concern here is that if we start additionally trying to request token, even if `allowedFlows` guarantees that we are not getting to the auth flow and just trying to refresh it, we now will have multiple separate errors to deal with on library consumer side when user is not authorized instead of just one clear "not authorized" network error

Reply:
Right, and I see that concern, which is why I’m suggesting that we document this and don’t actually change the behavior :slightly_smiling_face:

Reply:
Hmmm, actually I never checked that oidc-client will just propagate backend error instead of throwing something custom which will not be an expected outcome for atlas service. This makes less difference for oauth methods revoke and introspect, but maybe user info is just not something we can move here without making it too awkward to use

Reply:
You could expand `OIDCMockProvider` to support userinfo and then return errors from that and add a test that verifies that that error gets propagated (I’d be surprised if it wasn’t)

Reply:
Cool, will try to set it up and see how it looks like (and this is something I'll try anyway for the testing discussion we had)

Reply:
I bumped the story points on the ticket and moved it out of the project to the backlog as this will require us to implement the full user support for oidc mock provider. This increases the scope so much, I'm not sure it's worth doing. I'll close this PR for now